portal/src/graphql/portal/PasswordlessAuthenticatorScreen.tsx

```diff
@@ -0,0 +1,288 @@
+  props: PasswordlessAuthenticatorProps
+) {
+  const {
+    rawAppConfig,
+    emailAndSmsTemplates,
+    updateAppAndEmailSmsTemplatesConfig,
+    updatingAppAndEmailSmsTemplateConfig,
+  } = props;
+
+  const initialState: PasswordlessAuthenticatorScreenState = useMemo(() => {
+    return {
+      emailHtmlTemplate: emailAndSmsTemplates?.emailHtml ?? "",
+      emailPlainTextTemplate: emailAndSmsTemplates?.emailText ?? "",
+      smsTemplate: emailAndSmsTemplates?.smsText ?? "",
+    };
+  }, [emailAndSmsTemplates]);
+
+  const [state, setState] = useState<PasswordlessAuthenticatorScreenState>(
+    initialState
+  );
+
+  const isFormModified = useMemo(() => {
+    return !deepEqual(initialState, state, { strict: true });
+  }, [initialState, state]);
+
+  const onEmailHtmlTemplateChange = useCallback(
+    (_event: unknown, value: string | undefined) => {
+      if (value === undefined) {
+        return;
+      }
+      setState((state) => ({
+        ...state,
+        emailHtmlTemplate: value,
+      }));
+    },
+    []
+  );
+
+  const onEmailPlainTextTemplateChange = useCallback(
+    (_event: unknown, value: string | undefined) => {
+      if (value === undefined) {
+        return;
+      }
+      setState((state) => ({
+        ...state,
+        emailPlainTextTemplate: value,
+      }));
+    },
+    []
+  );
+
+  const onSaveButtonClicked = useCallback(() => {
+    if (rawAppConfig == null) {
+      return;
+    }
+
+    const newAppConfig = produce(rawAppConfig, (draftConfig) => {
+      draftConfig.template = draftConfig.template ?? {};
+      draftConfig.template.items = draftConfig.template.items ?? [];
+
+      if (
+        state.emailHtmlTemplate !== initialState.emailHtmlTemplate &&
+        !draftConfig.template.items.some(
+          (item) => item.type === EMAIL_HTML_TEMPLATE_NAME
+        )
+      ) {
+        draftConfig.template.items.push({
+          type: EMAIL_HTML_TEMPLATE_NAME,
+          uri: `file:///templates/${EMAIL_HTML_TEMPLATE_NAME}`,
+        });
+      }
+
+      if (
+        state.emailPlainTextTemplate !== initialState.emailPlainTextTemplate &&
+        !draftConfig.template.items.some(
+          (item) => item.type === EMAIL_TEXT_TEMPLATE_NAME
+        )
+      ) {
+        draftConfig.template.items.push({
+          type: EMAIL_TEXT_TEMPLATE_NAME,
+          uri: `file:///templates/${EMAIL_TEXT_TEMPLATE_NAME}`,
+        });
+      }
+
+      if (
+        state.smsTemplate !== initialState.smsTemplate &&
+        !draftConfig.template.items.some(
+          (item) => item.type === SMS_TEXT_TEMPLATE_NAME
+        )
+      ) {
+        draftConfig.template.items.push({
+          type: SMS_TEXT_TEMPLATE_NAME,
+          uri: `file:///templates/${SMS_TEXT_TEMPLATE_NAME}`,
+        });
+      }
+
+      clearEmptyObject(draftConfig);
+    });
+
+    // TODO: handle error
+    updateAppAndEmailSmsTemplatesConfig(newAppConfig, {
```

Comment:
This error should not needed to be handled (i.e. handled by parent component from the return value of hooks), so the TODO note is not needed.

Reply:
Removed